### PR TITLE
WRN-2399: Fixed Wizardpanel to not re-render the previous panel

### PR DESCRIPTION
**Pull request by jeonghee27**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
WRN-2399

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- Re apply https://github.com/enactjs/sandstone/pull/975
- Fix WizardPanels/useFocusOnTransition.js for WRN-2302 (Remove setstate structure)
- Divide the children hook(panel) and wrapping div that having `key`  (for WRN 2389)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRN-2399 PLAT-143483
WRN-2302 WRN-2389 

### Comments
Enact-DCO-1.0-Signed-off-by: Jeonghee Ahn (jeonghee27.ahn@lge.com)

### CODE REVIEW

File: WizardPanels/WizardPanels.js [hunk 1]
```diff
@@ -444,16 +444,17 @@ const WizardPanelsBase = kind({
...
-	let [reverse, setReverse] = useState(rtl);
-
-	if (prevIndex !== index) {
-		reverse = rtl ? (index > prevIndex) : (index < prevIndex);
-		setReverse(reverse);
-		setPrevIndex(index);
+	const prevIndex = useRef(index);
+	const reverse = useRef(rtl);
+	// If the index was changed, the panel transition is occured on the next cycle by `Panel`
+	const returnValue = {reverseTransition: reverse.current, prevIndex: prevIndex.current};
```
Comment (MikyungKim) [inline WizardPanels/WizardPanels.js:450]: ?
```suggestion
	const prev = {reverseTransition: reverse.current, prevIndex: prevIndex.current};
```

File: WizardPanels/useFocusOnTransition.js [hunk 2]
```diff
@@ -28,27 +30,7 @@ const transitionHandlers = {
...
-			current.timerId = setTimeout(() => {
-				const currentSpotlight = Spotlight.getCurrent();
-				if (config.spotlightId && !currentSpotlight) {
-					Spotlight.focus(config.spotlightId);
-				}
-			}, 16);
-		}
-
-		return () => clearTimeout(current.timerId);
-	}, [complete, config.spotlightId, focusIndex, current]);
```
Comment (MikyungKim) [inline WizardPanels/useFocusOnTransition.js:51]: Let's test without these dependencies on the board.

File: samples/sampler/stories/qa/WizardPanels.js [hunk 3]
```diff
@@ -177,10 +186,104 @@ LongPrevNextButtons.parameters = {
...
+
+WithChangingChildren.storyName = 'with changing children';
+WithChangingChildren.parameters = {
+	props: {
+		noPanel: true
+	}
+};
+
+class PureComponentItem extends PureComponent {
+
```
Comment (MikyungKim) [inline samples/sampler/stories/qa/WizardPanels.js:229]: Let's remove this line.

File: WizardPanels/useFocusOnTransition.js (hunk 2 above)
Comment (jeonghee27) [reply WizardPanels/useFocusOnTransition.js:51]: If we remove the dependecy, this use Effect called whenever WizardPanelsRouter called.
It is even called when a child is updated. please refer the "WithChangingChildren" sample

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**